include/h2o/memory.h

```diff
@@ -199,7 +199,17 @@ void h2o_mem_clear_pool(h2o_mem_pool_t *pool);
 /**
  * allocates given size of memory from the memory pool, or dies if impossible
  */
-void *h2o_mem_alloc_pool(h2o_mem_pool_t *pool, size_t sz);
+void *h2o_mem_alloc_pool_aligned(h2o_mem_pool_t *pool, size_t alignment, size_t size);
+
+#ifdef H2O_NO_BUILT_IN_ALIGNOF
```

Comment:
Please test the name of the compiler using `#if` here to determine if we can use `__alignof__`.

The header file will be used from other applications and that might not involve the invocation of CMake.

Reply:
Do not `determining it in CMakeFiles.txt` sounds good.

i searched on the web and seems most compilers support `__alignof__`.
if we use `#if` to judge, i'm confused about the judge condition.
is there any platform  we need support but missing `__alignof__`?

Reply:
That's a very good question.

I now think that we do not need a check. Let's expect that `__alignof__` is always available. We already use things like `__builtin_clzll`; it is likely that the range of compilers we support will remain the same.